rest/route/service.go

```diff
@@ -84,7 +84,8 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	app.AddRoute("/distros/{distro_id}/setup").Version(2).Patch().Wrap(editDistroSettings).RouteHandler(makeChangeDistroSetup(sc))
 	app.AddRoute("/distros/{distro_id}/teardown").Version(2).Get().Wrap(editDistroSettings).RouteHandler(makeGetDistroTeardown(sc))
 	app.AddRoute("/distros/{distro_id}/teardown").Version(2).Patch().Wrap(editDistroSettings).RouteHandler(makeChangeDistroTeardown(sc))
-	app.AddRoute("/distros/{distro_id}/execute").Version(2).Post().Wrap(editDistroSettings).RouteHandler(makeDistroExecute(sc, env))
+	app.AddRoute("/distros/{distro_id}/execute").Version(2).Patch().Wrap(editDistroSettings).RouteHandler(makeDistroExecute(sc, env))
+	app.AddRoute("/distros/{distro_id}/icecream_config").Version(2).Patch().Wrap(editDistroSettings).RouteHandler(makeDistroIcecreamConfig(sc, env))
```

Comment:
shouldn't execute and icecream_config have a different/greater permissions?

Reply:
There's no better match on permissions because we deleted superusers. This is the closest matching permission level - if you can change a distro, you have power over hosts on this distro.

Reply:
I changed this to require permission to edit this host (which is the same permission level that lets you edit the host status).